ablySpec/RealtimeClientChannel.swift

```diff
@@ -13,6 +13,63 @@ class RealtimeClientChannel: QuickSpec {
     override func spec() {
         describe("Channel") {
 
+            // RTL3
+            context("connection state") {
+
+                // RTL3a
+                pending("changes to FAILED") {
+
+                    it("ATTACHING channel should transition to FAILED") {
+                        let options = AblyTests.commonAppSetup()
+                        options.autoConnect = false
+                        let client = ARTRealtime(options: options)
+                        client.setTransportClass(TestProxyTransport.self)
+                        client.connect()
+                        defer { client.close() }
+
+                        let channel = client.channel("test")
+                        channel.attach()
+                        let transport = client.transport as! TestProxyTransport
+                        transport.actionsIgnored += [.Attached]
+
+                        expect(channel.state).to(equal(ARTRealtimeChannelState.Attaching))
+
+                        waitUntil(timeout: testTimeout) { done in
+                            channel.subscribeToStateChanges { state, status in
+                                if state == .Failed {
+                                    expect(status.state).to(equal(ARTState.Error))
+                                    expect(status.errorInfo!.code).to(equal(90000))
+                                    done()
+                                }
+                            }
+                            client.onError(AblyTests.newErrorProtocolMessage())
+                        }
+                    }
+
+                    it("ATTACHED channel should transition to FAILED") {
+                        let client = ARTRealtime(options: AblyTests.commonAppSetup())
+                        defer { client.close() }
+
+                        let channel = client.channel("test")
+                        channel.attach()
+                        expect(channel.state).toEventually(equal(ARTRealtimeChannelState.Attached), timeout: testTimeout)
+
+                        waitUntil(timeout: testTimeout) { done in
+                            channel.subscribeToStateChanges { state, status in
+                                if state == .Failed {
+                                    expect(status.state).to(equal(ARTState.Error))
+                                    expect(status.errorInfo!.code).to(equal(90000))
```

Comment:
Might be nice in both of these tests to simply double check that the channel state itself as failed as opposed to simply emitting the failed state.  I appreciate this is probably covered elsewhere, but the test explicitly states that the channel should transition to failed, and as such, it's nice to check that it both emits a failed state and has a failed state.